File: leetcode/226/solution.py
"""
https://leetcode.com/problems/invert-binary-tree/
226. Invert Binary Tree
Given the root of a binary tree, invert the tree, and return its root.
"""
from typing import Optional

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root:
            root.left, root.right = root.right, root.left
            self.invertTree(root.left)
            self.invertTree(root.right)
        return root

File: leetcode/226/solution.go
/*
https://leetcode.com/problems/invert-binary-tree/
226. Invert Binary Tree
Given the root of a binary tree, invert the tree, and return its root.
*/
package main

// Definition for a binary tree node.
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func invertTree(root *TreeNode) *TreeNode {
	if root != nil {
		root.Left, root.Right = root.Right, root.Left
		invertTree(root.Left)
		invertTree((root.Right))
	}
	return root
}
